packages/boot/test/bootstrapTests/benchmarkerator.md

```diff
@@ -0,0 +1,171 @@
+and only if there is demand for it.  We are not currently planning to adapt it
+for non-swingset benchmarking, though it is possible that a similar tool may be
+concocted for that purpose should it seem warranted.
+
+[^1]: The term "chain" is slightly confusing here, but will have to do until the
+    product folks give us better terminology.  There is no actual chain per se,
+    merely a swingset containing the various vats and contracts that make up the
+    default Agoric ecosystem, along with an ersatz bridge device that can be
+    used to inject message traffic as if it had originated from actual Cosmos
+    messages.
+
+## Writing a benchmark
+
+By convention a benchmark should be placed in source file a named
+`benchmark-yourbenchmarkname.js` (substitute your own benchmark name of course),
+which, by convention, goes in the `benchmarks` directory of your package,
+parallel to the conventional `test` directory (as of this writing very few
+packages have such a `benchmarks` directory, so you might need to add it
+yourself).[^2]
+
+[^2]: In the fullness of time, the `yarn bench` command will be updated to run
+    all the benchmarks found in the `benchmarks` directory, much as it now runs
+    the older, Ava-based benchmarks that it looks for in the `test` directory.
+    Those will themselves be deprecated and/or moved into `benchmarks`.
+    However, as of this writing that has not yet been done.
+
+The first thing a benchmark should do is import The Benchmarkerator:[^3]
+```
+import { bench } from '@agoric/benchmark';
+```
+
+[^3]: As of this writing The Benchmarkerator actually lives in
+    `@agoric/boot/test/boostrapTests/benchmarkerator.js`.  This should be
+    changed soon, hopefully before the first PR containing this writeup is
+    landed, in which case you will never see this footnote.
+
+Note that this importation usually should be the very first thing you do, much
+as you typically import `@agoric/swingset-vat/tools/prepare-test-env-ava.js` or
+`@agoric/zoe/tools/prepare-test-env-ava.js` or the like as the first thing in a
+test implementation.  The exception, as with tests, is imports that bring in any
+kind of pre-lockdown functionality that you will be needing to use; however,
+that should be necessary only in very specialized circumstances.
+
+In a manner similar to the way tests are packaged for Ava, a benchmark file can
+actually contain multiple benchmark tests.  Each should be declared with a
+statement of the form:
+
+`bench.addBenchmark(label: string, benchmark: Benchmark);`
+
+where `label` is a string labelling the benchmark (much as individual tests are
+labelled in Ava), and `benchmark` is an object describing the benchmark itself
+(which will be described in detail below)
+
+After defining one or more benchmarks with `addBenchmark` you must then end by
+invoking:
+
+```await bench.run(name);```
+
+where `name` is a string naming the benchmark run for purposes of error and
+result reporting.
```

Comment:
The primary user benefit is that it's self contained.  But the main reason is that it makes deployment a lot easier for us (or for me, anyway) at least at the early stage we're at now, because it's just an import rather than something that has be published as its own npm package and then installed.